packages/docs/versioned_docs/version-2.0.0-alpha-3/welcome.md

```diff
@@ -8,21 +8,27 @@ title: What is UniRep?
 
 ## Overview
 
-**UniRep** (**Uni**versal **Rep**utation) is a _private_ and _non-repudiable_ **reputation system**. Users can&#x20;
+**UniRep** (**Uni**versal **Rep**utation) is a _private_ and _non-repudiable_ **reputation system**. Users can:&#x20;
 
 1. Receive positive and negative reputation from attesters.
-2. Voluntarily prove that they have at least certain amount of reputation without revealing the exact amount.&#x20;
-3. Moreover, users cannot refuse to receive reputation from an attester.
+2. Voluntarily prove that they have at least a certain amount of reputation without revealing the exact amount.&#x20;
+3. Users cannot refuse to receive reputation from an attester.
 
-The high-level goal for **UniRep** is to be a base layer on top of which anyone can easily build _custom_, yet _interoperable_, reputation systems, For instance, users could create combined zero-knowledge proofs of reputation across different social media platforms, consumer apps, or financial applications, in order to provide holistic, private, and trustworthy information about themselves to others.
+The high-level goal for **UniRep** is to be a base layer on top of which anyone can easily build _custom_, yet _interoperable_, reputation systems. For instance, users could create combined zero-knowledge proofs of reputation across different social media platforms, consumer apps, or financial applications in order to provide holistic, private, and trustworthy information about themselves to others.
+
+TODO: link to new Medium article from Althea and Ki Chong [here]
 
 UniRep is originally proposed by BarryWhiteHat in [this ethresear.ch post](https://ethresear.ch/t/anonymous-reputation-risking-and-burning/3926)
 
-## v2
+## Implementation
+
+TODO: brief technical overview
```

Comment:
@vimwitch @vivianjeng any thoughts here?

Reply:
yeah I think we can have some technical implementation overview here
I think we can explain each feature from its technical perspective

1. Receive positive and negative reputation from attesters.
With Unirep protocol, we use **Semaphore identity** to generate a temporary identity called **epoch key**, which is a hash of identity, epoch and a nonce. Users can provide this **epoch key** to receive positive and negative reputation. Like sending a token to an Ethereum address, **epoch key** is the address to the Unirep reputation account.

2. Voluntarily prove that they have at least a certain amount of reputation without revealing the exact amount.
[Zero-knowledge proofs](https://ethereum.org/en/zero-knowledge-proofs/) is used in Unirep protocol. That means users don't have to provide the private information of their reputation balance but the balance can be verified in the proof. User can claim a minimum reputation that the user wants to prove, then the unirep circuits will check if `(posRep-negRep) >= minRep`.
> `posRep`: total positive reputation the user has
> `negRep`: total negative reputation the user has
> `minRep`: the minimum reputation the user claim to have

3. Users cannot refuse to receive reputation from an attester.
After an epoch finishes, the user should perform an action called **user state transition** to receive reputation so that the user won't be able to refuse any of the epoch keys he has. We use **epoch trees** to achieve this.
**Epoch tree** is a sparse merkle tree where the leaf value is the hash of reputation balance and the leaf index is the epoch key.
If one of reputation balance changes, the epoch tree root will be totally different as well.
So in the user state transition step, the user will generate a zero-knowledge proof to prove that
1) the user owns `n` epoch keys
2) each epoch key has the balance that matches **epoch tree**
3) user updates its total balance by e.g. `oldPosRep + newPosRep` (where `oldPosRep` is total positive reputation accumulated from user sign up and `newPosRep` is the total positive reputation in the transition epoch)